### ISSUE TITLE
HelathCare: Patient Name should be Customer's name

### ISSUE THREAD

**Issue by Helkyd** [frappe/erpnext#11021]
As i have commented https://discuss.erpnext.com/t/erpnext-healthcare-merged-to-develop/28205/20
Also i have noticed that by default Administrator cannot delete the Patient record (correct) but don't see also the Status option to disable the record if the same is not correct.

I have created two Patients with same name and on the first the Customer name was the Patient record name and on the Second before creating the Patient i created the Customer in order to select it when creating the Patient .... But ended up with two Patients with same name and no option to disable the record with Customer name as Record number.

**Comment by Zlash65** [comment]
![patient-name](https://user-images.githubusercontent.com/11695402/31995750-5d555764-b9a3-11e7-9d9d-9b040f33eb9e.gif)
Will this serve the purpose ? 
Saving the Patient will create Customer with the same name and that customer will have a link back to the Patient entry. Its a two way link between both and in a closed loop yet [ deleting either of them would be not possible. As such deleting a patient or a customer from System is really not necessary right? ]

**Comment by Helkyd** [comment]
Agree. Deleting is not advised as the info is always required. But you should be able to disable the Patient and not the customer as I might need to buy something from the pharmacy as customer.

**Comment by mascor** [comment]
Hi,
is the function of automatically creating a customer with the same name of the patient, already implemented in the last version of ERPNext?

Thanks
Bye

**Comment by palashjhabak** [comment]
@Zlash65 I feel that approach is not generic enough because in a big company scenario a person can already be a customer of that company but may not be a patient yet and when he becomes one, there will be duplicate customer. There could be scenario where company sells other services and products around health industries e.g fitness classes, supplements or a hospital could be integrated with a medical college and have a lot of different set of customers already some of whom may become a patient in future. I think this is better solved in the case of Employee and System user where in Employee has to a system user first and we can select a system user while creating a new employee, the same can be followed here. This is what I feel, please correct me if I am wrong. 

Thanks

**Comment by system19** [comment]
By having a seperate Patient name and linked Customer name allows a clinic to have a Patient but bill an insurance company or a state authority. Maybe children under 5 might get Govt sponsored immunisation as an example. I don't think that Patient medical records should be able to be deleted easily but see no reason why Customer info needs to be so stringent.

**Comment by palashjhabak** [comment]
@Zlash65 I think this will create even more accidental duplicates if customer field is not mandatory and a new customer is created on save. So it should be a strict dependency as in any case being a patient is just a subset of being a customer, for, a hospital might have other products and services for customers who are non patients e.g Yoga or Fitness classes, Workshops for Medical Students, Supplement Subscriptions etc.

**Comment by palashjhabak** [comment]
>           Hi,
> is the function of automatically creating a customer with the same name of the patient, already implemented in the last version of ERPNext?
> Thanks
> Bye

@mascor From what I see currently in the last version (v10) what it does is it creates a patient first and then use that auto generated Patient ID as the name of the new customer it generates. I think this flow it totally flawed. As I mentioned above it should be the other way round - Being a patient is just a subset of being a Customer

**Comment by system19** [comment]
@mascor Good point.
